gtsam/hybrid/tests/testHybridFactorGraph.cpp

```diff
@@ -376,67 +361,271 @@ TEST(HybridFactorGraph, ToDecisionTreeFactor) {
+//// Test if we can incrementally do the inference
+//TEST_UNSAFE(DCGaussianElimination, Incremental_inference) {
+//  Switching three_step(3);
+//
+//  // We want to eliminate x1, x2 and x3
+//  Ordering ordering;
+//  ordering += X(1);
+//  ordering += X(2);
+//  ordering += X(3);
+//
+//  // Eliminate the factor graph and get the
+//  HybridBayesNet::shared_ptr hybridBayesNet;
+//  HybridFactorGraph::shared_ptr remainingFactorGraph;
+//  std::tie(hybridBayesNet, remainingFactorGraph) =
+//      three_step.linearizedFactorGraph.eliminatePartialSequential(ordering);
+//
+//  hybridBayesNet->print("hybridBayesNet");
+//  remainingFactorGraph->print("rfg");
+//
+//  Switching four_step(4);
+//
+//  HybridFactorGraph hf;
+//
+////  for (const auto& f : remainingFactorGraph->discreteGraph())
+////    hf.push_discrete(f);
+//
+////  auto conditional = boost::make_shared<DiscreteConditional>(
+////      DiscreteKey{M(3), 2}, DiscreteKeys{{M(2), 2}}, "1/2 3/2");
+////  hf.push_discrete(conditional);
+//
+//  hf.push_dc(four_step.nonlinearFactorGraph.dcGraph().at(2));
+//
+//  Values lp;
+//  lp.insert(X(3), four_step.linearizationPoint.at(X(3)));
+//  lp.insert(X(4), four_step.linearizationPoint.at(X(4)));
+//  auto lhf = hf.linearize(lp);
+//
+//  lhf.push_dc(hybridBayesNet->at(2)); // Density on x3
+//
+//  GTSAM_PRINT(lhf);
+//
+//  ordering.clear();
+//  ordering += X(3);
+//  ordering += X(4);
+//
+//  // Eliminate the factor graph and get the
+//  HybridBayesNet::shared_ptr hybridBayesNet_4;
+//  HybridFactorGraph::shared_ptr remainingFactorGraph_4;
+//  std::tie(hybridBayesNet_4, remainingFactorGraph_4) =
+//      lhf.eliminatePartialSequential(ordering);
+//
+//  hybridBayesNet_4->print("hybridBayesNet_4:");
+//  remainingFactorGraph_4->print("rfg4");
+//
+//  remainingFactorGraph->discreteGraph().product().print("prod");
+//}
+
+class IncrementalHybrid {
+
+ public:
+
+  HybridBayesNet::shared_ptr hybridBayesNet_;
+  HybridFactorGraph::shared_ptr remainingFactorGraph_;
+  /**
+   * Given new factors, perform an incremental update.
+   * @param graph The new factors, should be linear only
+   */
+  void update(HybridFactorGraph graph, const Ordering &ordering) {
+    // TODO(fan): add all factors involving variables in the `ordering`
+
+    if (ordering.at(0) == X(2)) graph.push_back(hybridBayesNet_->at(1));
+    // Eliminate partially.
+    std::tie(hybridBayesNet_, remainingFactorGraph_) =
+        graph.eliminatePartialSequential(ordering);
+  }
 
-  CHECK(result.second);
-  //  GTSAM_PRINT(*result.second);  // HybridFactorGraph
-  EXPECT_LONGS_EQUAL(3, result.second->size());
+};
+
+/* ****************************************************************************/
+// Test if we can incrementally do the inference
+TEST_UNSAFE(DCGaussianElimination, Incremental_inference) {
+  Switching switching(3);
+
+  IncrementalHybrid incrementalHybrid;
+
+  HybridFactorGraph graph1;
```

Comment:
I think we need the graph0 stage now as we discussed. It threw an error, probably because we made a wrong assumption somewhere?

Reply:
Yes, maybe after merging this?